library_generation/generate_library.sh

```diff
@@ -280,5 +295,30 @@ popd # output_folder
 #####################################################
 pushd "${output_folder}/${destination_path}"
 rm -rf java_gapic_srcjar java_gapic_srcjar_raw.srcjar.zip java_grpc.jar java_proto.jar temp-codegen.srcjar
-popd
-set +x
+popd # destination path
+##################### Section 5 #####################
+# post-processing
+#####################################################
+source "${script_dir}/post_processing/postprocessing_utilities.sh"
+if [ "${enable_postprocessing}" != "true" ];
+then
+  echo "post processing is disabled"
+  exit 0
+fi
+if [ -z "${versions_file}" ];then
+  echo "no versions.txt argument provided. Please provide one in order to enable post-processing"
```

Comment:
It's OK to make it required for now since it's only for existing libraries. For new client libraries, I'm assuming we have to generate versions.txt first before calling this script? Is it generated by new client library script currently? cc @JoeWang1127

Reply:
@blakeli0 the version of a new library [defaults](https://github.com/googleapis/synthtool/blob/247239027cad7678adea8f57e529ebd12600ee93/docker/owlbot/java/src/fix-poms.py#L333) to `0.0.1-SNAPSHOT`.

[example PR](https://github.com/googleapis/google-cloud-java/pull/9993/files)

The new library script [applies current versions](https://github.com/googleapis/google-cloud-java/blob/389ac8dca201302a2ceb2299b32eb69fd0d79077/generation/new_client/new-client.py#L364) but this is for other referenced artifacts such as `google-cloud-java` ([example](https://github.com/googleapis/google-cloud-java/blob/9bf5aae4202c5e6c7c3e23e0ae1d85be849b3a9c/java-redis-cluster/google-cloud-redis-cluster-bom/pom.xml#L12))

I think we still need a versions file for referenced artifacts other than the library being generated when dealing with a new monorepo library